scripts/scraper-ng/rules/html-require-recipe-ingredients/ingredient-handlers/data-constructor.js

```diff
@@ -0,0 +1,94 @@
+const select = require("hast-util-select");
+
+const utils = require("./utils.js");
+const checkLinkList = require("./link-list-checker.js");
+
+function checkText(node, text) {
+  return node.type === "text" && node.value === text;
+}
+
+function checkTag(node, tag) {
+  return node.type === "element" && node.tagName === tag;
+}
+
+/**
+ * Handler for the `data.constructor` ingredient.
+ */
+function handleDataConstructor(tree, logger) {
+  // Constructor is a mandatory property
+  const heading = select.select(`h2#Constructor`, tree);
+  if (heading === null) {
+    logger.expected(tree, `h2#Constructor`, "expected-heading");
+    return null;
+  }
+
+  // Check common link list structure
+  let ok = checkLinkList("Constructor", tree, logger);
+
+  // This link list is only allowed one entry
+  const section = utils.sliceSection(heading, tree);
+  const dts = select.selectAll("dt", section);
+  if (dts.length !== 1) {
+    logger.fail(
+      section,
+      "Constructor section may only contain one DT item",
+      "only-single-constructor-dt"
+    );
+    ok = false;
+  }
+  const dds = select.selectAll("dd", section);
+  if (dds.length !== 1) {
+    logger.fail(
+      section,
+      "Constructor section may only contain one DD item",
+      "only-single-constructor-dd"
+    );
+    return null;
+  }
+
+  // The dd must be of the form: "Creates a new <a>...</a> object."
+  const dd = dds[0];
+  if (dd.children.length !== 3) {
+    logger.fail(
+      section,
+      "Constructor description must contain three nodes",
```

Comment:
The messages here are kinda particular to the way hast represents the HTML. I'm going to suggest some alternatives

```suggestion
      "Constructor description must be in the form `Creates a new <a>...</a> object.`",
```